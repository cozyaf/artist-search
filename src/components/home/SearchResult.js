import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function SearchResult({ searchResult }) {
  const { id, name, disambiguation } = searchResult;

  return (
    <>
      <Link to={ `/artist/${id}` } style={{ textDecoration: 'none' }}>
        <h2>{name}</h2>
        <p>{disambiguation ? disambiguation : 'description does not exist'}</p>
      </Link>
    </>
  );
}

SearchResult.propTypes = {
  searchResult: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    disambiguation: PropTypes.string
  }).isRequired
};

export default SearchResult;
