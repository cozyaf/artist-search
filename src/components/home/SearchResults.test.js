import React from 'react';
import { shallow } from 'enzyme';
import SearchResult from './SearchResult';

describe('SearchResult component', () => {
  it('renders SearchResult', () => {
    const wrapper = shallow(<SearchResult searchResult={{ id:'', name:'', disambiguation:'' }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
