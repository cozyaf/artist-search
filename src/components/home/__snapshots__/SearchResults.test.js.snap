// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component renders SearchResult 1`] = `
<Fragment>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/artist/"
  >
    <h2 />
    <p>
      description does not exist
    </p>
  </Link>
</Fragment>
`;
